### PR TITLE
[release-1.5] Update envoy sha rel15

### PR DESCRIPTION
**Pull request by jacob-delgado**
**What this PR does / why we need it**:

Update proxy sha based on Istio 1.5.7 security patch fix

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -38,21 +38,21 @@ bind(
 # 2. Update .bazelversion, envoy.bazelrc and .bazelrc if needed.
 #
 # istio/envoy commit date: Jun 02 2020
```
Comment (shamsher31) [inline WORKSPACE:40]: nit: Is it require to update this date also if the SHA is updated?

Ref: Last PR https://github.com/istio/proxy/pull/2867/files#diff-09498dbadf45966909850dc8a47ebb13R40

Comment (jacob-delgado) [reply WORKSPACE:40]: It was an oversight, but given that it's just a comment and it already got merged it's fine. Making sure the correct sha is updated is more important. Didn't notice that.

**Pull request merged**